src/pages/home/report/ReportActionsList.js

```diff
@@ -170,25 +179,41 @@ function ReportActionsList({
             return;
         }
 
+        cacheUnreadMarkers.delete(report.reportID);
         reportActionSize.current = sortedReportActions.length;
         setCurrentUnreadMarker(null);
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [sortedReportActions.length, report.reportID]);
 
     useEffect(() => {
-        const didManuallyMarkReportAsUnread = report.lastReadTime < DateUtils.getDBTime() && ReportUtils.isUnread(report);
-        if (didManuallyMarkReportAsUnread) {
-            // Clearing the current unread marker so that it can be recalculated
-            setCurrentUnreadMarker(null);
-            setMessageManuallyMarkedUnread(new Date().getTime());
+        if (!userActiveSince.current || report.reportID !== prevReportID.current) {
             return;
         }
-
+        if (!messageManuallyMarkedUnread && lastReadRef.current && lastReadRef.current < report.lastReadTime) {
+            cacheUnreadMarkers.delete(report.reportID);
+        }
+        lastReadRef.current = report.lastReadTime;
         setMessageManuallyMarkedUnread(0);
 
-        // We only care when a new lastReadTime is set in the report
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [report.lastReadTime]);
+    }, [report.lastReadTime, report.reportID]);
+
+    useEffect(() => {
+        // If the reportID changes, we reset the userActiveSince to null, we need to do it because
+        // this component doesn't unmount when the reportID changes
+        if (unreadActionSubscription.current) {
+            unreadActionSubscription.current.remove();
+            unreadActionSubscription.current = null;
+        }
+
+        // Need to listen for the specific reportID, otherwise we could be listening to all the reports
+        unreadActionSubscription.current = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+            cacheUnreadMarkers.delete(report.reportID);
+            lastReadRef.current = newLastReadTime;
+            setCurrentUnreadMarker(null);
+            setMessageManuallyMarkedUnread(new Date().getTime());
```

Comment:
Was added [here](https://github.com/Expensify/App/pull/26173) point 3. 
I just moved the code around because of conflicts. Basically setting a new number produces a rerender, but when is clean, setting 0, and we try to set 0 again, because it is 0 already no rerender event is fired